src/OpenCensus/Utils/Arrays.cs

```diff
@@ -111,11 +111,16 @@ internal static char[] ByteToHexCharArray(byte b)
             return result;
         }
 
-        internal static byte[] StringToByteArray(string src)
+        internal static byte[] StringToByteArray(string src, int start = 0, int len = -1)
         {
-            int size = src.Length / 2;
+            if (len == -1)
+            {
+                len = src.Length;
+            }
+
+            int size = len / 2;
             byte[] bytes = new byte[size];
-            for (int i = 0, j = 0; i < size; i++)
+            for (int i = 0, j = start; i < size; i++)
             {
                 int high = HexCharToInt(src[j++]);
```

Comment:
this seems to be extremely inefficient with all allocations it does. Here is a wonderful perf analysis comparison for different algos to convert byte array to string https://stackoverflow.com/questions/311165/how-do-you-convert-a-byte-array-to-a-hexadecimal-string-and-vice-versa

we should do this: https://stackoverflow.com/questions/311165/how-do-you-convert-a-byte-array-to-a-hexadecimal-string-and-vice-versa/24343727#24343727

Reply:
created https://github.com/census-instrumentation/opencensus-csharp/issues/53